the mksquashfs code has an all_zeros func that does:
	long *p = (long *) file_buffer->data;

make sure we force the data field to be aligned so that still works.

--- a/squashfs-tools/caches-queues-lists.h
+++ b/squashfs-tools/caches-queues-lists.h
@@ -123,6 +123,7 @@ struct file_buffer {
 	char locked;
 	char wait_on_unlock;
 	char noD;
+	long _pad;
 	char data[0];
 };
 
